### ISSUE TITLE
Python Function Raises Exception Importing pyodbc

### ISSUE THREAD

**Issue by jkytola-nh** [Azure/static-web-apps#291]
The Build and Deploy Job successfully installs pyodbc via requirements.txt

`Successfully installed azure-functions-1.6.0 pyodbc-4.0.30`

However, when executed, the function returns a 500 error. App Insights shows the following:

```
Exception while executing function: Functions.Example Result: Failure
Exception: ImportError: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /home/site/wwwroot/.python_packages/lib/site-packages/pyodbc.cpython-38-x86_64-linux-gnu.so)
Stack:   File "/azure-functions-host/workers/python/3.8/LINUX/X64/azure_functions_worker/dispatcher.py", line 355, in _handle__invocation_request
    call_result = await self._loop.run_in_executor(
  File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/azure-functions-host/workers/python/3.8/LINUX/X64/azure_functions_worker/dispatcher.py", line 542, in __run_sync_func
    return func(**params)
  File "/home/site/wwwroot/Example/__init__.py", line 7, in main
    import pyodbc
```

Example repo reproducing this issue available here

https://github.com/jkytola-nh/static-app-quickstart

**Comment by anthonychu** [comment]
I can repro this. Looks like Oryx is not building the function app with the packages restored to `.python_packages/lib/site-packages`.

Repro: https://github.com/anthonychu/20201210-test-python-swa

This app has a zip endpoint that allows me to download the function app package. The folder is missing.

/cc @miwebst @mkarmark

**Comment by erfannariman** [comment]
Is there any way workaround for this for now? It broke our fine running functions with new deployments and we have no idea how to solve this. @anthonychu

**Comment by erfannariman** [comment]
I found a solution: 

For those who are using GitHub actions for deployment, set the ubuntu version to 18.04:

```yaml
jobs:
  build-and-deploy-to-Azure:
    runs-on: ubuntu-18.04
```

**Comment by anthonychu** [comment]
@erfannariman It's interesting that this fixed it for you. It didn't fix the issue for me. The Static Web Apps GitHub Action runs in a container, so what you set as the OS for the outer GitHub Actions agent shouldn't affect the containerized build and deploy environment. We've identified that there's a potential mismatch between the Static Web Apps GitHub Action container (Debian Stretch) and the Azure Function Python environment (Buster). We'll work on updating the build container to Buster as well.

**Comment by anthonychu** [comment]
Here's a workaround while we sort out the problem with building in the Static Web Apps action.

Insert [these steps](https://github.com/anthonychu/20201210-test-python-swa/blob/df092d24a8f2f7747762b6efe90560b167290a6e/.github/workflows/azure-static-web-apps-agreeable-field-04b77ec1e.yml#L21-L30) to your action, and set the build agent to `ubuntu-18.04`. This builds the Python function app correctly first, then uses the Static Web Apps action to deploy it.

```yaml
      - name: Setup Python
        uses: actions/setup-python@v2
        with:
          # Version range or exact version of a Python version to use, using SemVer's version range syntax.
          python-version: "3.8.x"
      - name: Install dependencies
        run: |
          python -m pip install --target=".python_packages/lib/site-packages" -r requirements.txt
          echo "" > requirements.txt # blank out requirements.txt so it doesn't get installed again
        working-directory: api
```

**Comment by Kapkan-Lesion** [comment]
Hi @anthonychu, I have a function app that was running into a similar issue where pyodbc module was not found, despite the fact that I have it in the requirements.txt, and made sure to manually pip install this to double check. 

I managed to make it work with func azure functionapp publish <function app name>, it started working for some time...

Then boom, everything went down, it was working for a bit, but now the container can't even start up, I tried to create a dummy python function in portal, then deploy the same code to it, container emits the same behavior, it may start for a sec, then go down completely, or simply just will not start...

What's funny is that these functions all carry the same source code from a function that was working flawlessly...

Function name: 
`PythonTestMara10 `(the one where container basically gives up for some reason)
`pythonmarafeb27` (The one where function shows runtime unreachable, despite no changes being made to source code, the only difference I can think of would be this one is on Function extension runtime version ~3, whereas the one above is on ~2).

Do you mind having a quick look at what's happening? I cannot seem find relevant clues on why the container fails to start, which makes this a pain in locating the root cause...

Thank you!

**Comment by anthonychu** [comment]
@Kapkan-Lesion While the symptoms and possibly the underlying issues are the same, if you're using Azure Functions without Azure Static Web Apps, please use this issue to track this problem: https://github.com/Azure/azure-functions-python-worker/issues/818

/cc @Hazhzeng

**Comment by anthonychu** [comment]
@jkytola-nh If you still have this project, can you please give it a try again? Seems like I can't reproduce this anymore.

**Comment by jkytola-nh** [comment]
Confirmed - import is now successful. Thanks!

**Comment by erfannariman** [comment]
@anthonychu @jkytola-nh just to be sure, when you say "can't reproduce anymore", do you mean with the workaround, given in this comment (https://github.com/Azure/static-web-apps/issues/291#issuecomment-787400799) or something else? Since I just got the error when deploying an Azure Function App.

**Comment by jkytola-nh** [comment]
@erfannariman No workaround. A newly deployed function from the original example repo no longer has this issue.

**Comment by Hazhzeng** [comment]
Hi @erfannariman, @jkytola-nh, thanks for raising this issue. I need to collect some information from your deployment.
Are you using a pipeline to deploy the function app? If so, which vmImage are you using, ubuntu-18.04, ubuntu-20.04, or latest?

**Comment by erfannariman** [comment]
> Hi @erfannariman, @jkytola-nh, thanks for raising this issue. I need to collect some information from your deployment.
> Are you using a pipeline to deploy the function app? If so, which vmImage are you using, ubuntu-18.04, ubuntu-20.04, or latest?

I used ubuntu-latest, which will resolve to ubuntu-20.04 if I am not mistaken. Setting image to ubuntu-18.04 resolved the issues.

**Comment by Hazhzeng** [comment]
I see, thanks for reporting this @erfannariman, the engineering team will be acknowledged.

**Comment by erfannariman** [comment]
I forgot to mention I use GH actions for our deployment @Hazhzeng

**Comment by anthonychu** [comment]
@Hazhzeng Keep in mind that the Static Web Apps action runs in a Debian 9/stretch container. The version of the agent shouldn't have an effect. We plan on updating the image to 10/buster which should align more closely with the recommendation of ubuntu-18.04 for building Python function apps.

**Comment by vincda** [comment]
Issue just reproduced for Python 3.8 and pyodbc 4.0.30, with Github actions for deployment. Could fix it using Ubunbtu-18.04 instead of unbuntu-latest as mentioned by @erfannariman.
